### ISSUE TITLE
The EXPECTEDFAIL_NOALIAS in `cs_tests/recur9.c` should be NOALIAS?

### ISSUE THREAD

**Issue by taquangtrung** [SVF-tools/Test-Suite#28]
Hi,

In this test case, I think the assertion `EXPECTEDFAIL_NOALIAS(b,&b1)` should be `NOALIAS(b,&b1)`?

It is very similar to the assertion `NOALIAS(c,&c1)`, which I think is correct.

```
// cs_tests/recur9.c

#include "aliascheck.h"
int z;
void foo(int **a);
void bar(int **q){
    int** a = malloc(10);
    foo(a);
}

void foo(int **a){
    *a = &z;
    bar(a);
}

int main(){
    int** a,*b,*c,b1,c1;
    b = &b1;
    a = &b;
    foo(a);
    MUSTALIAS(b,&z);
    EXPECTEDFAIL_NOALIAS(b,&b1);
    a = &c;
    foo(a);
    MUSTALIAS(c,&z);
    NOALIAS(c,&c1);
}
```

**Comment by yuleisui** [comment]
Conservative underlying analysis could be improved.

**Comment by taquangtrung** [comment]
Do you suggest that both `EXPECTEDFAIL_NOALIAS` and `NOALIAS` are acceptable for `b` and `&b1`?

**Comment by taquangtrung** [comment]
Hi again,

Sorry for bothering you a lot this morning.
I think this test case also needs to be fixed for the non-terminating recursive call...

Thank you a lot again for fixing them! :-)

**Comment by yuleisui** [comment]
Done and thank you~
f8ac87128bd29fc281ea96cf13b3307fa0808385

**Comment by taquangtrung** [comment]
For a quick update, when a base case is added to `foo`, I think the two assertions `MUSTALIAS` will become `MAYALIAS` (when the call to `foo` terminates immediately at the base case).

How do you think about it?

**Comment by yuleisui** [comment]
Yes, correct. However, for current SVF's flow-sensitive implementation, we did not distinguish MAY and MUST for now. For your usage purposes, I have changed it.

**Comment by taquangtrung** [comment]
Thank you very much! :-)
